airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py

```diff
@@ -52,48 +58,52 @@ def send_request(self, query: str) -> SearchGoogleAdsResponse:
     @staticmethod
     def get_fields_from_schema(schema: Mapping[str, Any]) -> List[str]:
         properties = schema.get("properties")
-
         return [*properties]
 
     @staticmethod
-    def convert_schema_into_query(schema: Mapping[str, Any], report_name: str, from_date: str, to_date: str) -> str:
+    def convert_schema_into_query(
+        schema: Mapping[str, Any], report_name: str, from_date: str = None, to_date: str = None, cursor_field: str = None
+    ) -> str:
         from_category = REPORT_MAPPING[report_name]
         fields = GoogleAds.get_fields_from_schema(schema)
         fields = ",\n".join(fields)
 
-        query = Template(
-            """
-          SELECT
-            $fields
-          FROM $from_category
-          WHERE segments.date > '$from_date'
-            AND segments.date < '$to_date'
-          ORDER BY segments.date
-      """
-        )
-        query = query.substitute(fields=fields, from_category=from_category, from_date=from_date, to_date=to_date)
+        query_template = "SELECT {fields} FROM {from_category} "
+        substitute_params = dict(fields=fields, from_category=from_category)
+
+        if cursor_field:
+            query_template += "WHERE {cursor_field} > '{from_date}' AND {cursor_field} < '{to_date}' ORDER BY {cursor_field}"
+            substitute_params.update(dict(from_date=from_date, to_date=to_date, cursor_field=cursor_field))
 
+        query = query_template.format(**substitute_params)
         return query
 
     @staticmethod
-    def get_field_value(result: GoogleAdsRow, field: str) -> str:
+    def get_field_value(field_value: GoogleAdsRow, field: str) -> str:
         field_name = field.split(".")
-        try:
-            field_value = result
-            for level_attr in field_name:
-                field_value = field_value.__getattr__(level_attr)
-                if isinstance(field_value, Enum):
-                    field_value = field_value.name
+        for level_attr in field_name:
+            try:
+                field_value = getattr(field_value, level_attr)
+            except AttributeError:
+                field_value = getattr(field_value, level_attr + "_", None)
```

Comment:
Why do we have here such logic with try/except? The name of the attribute is unknown (I mean whether `field_value` has an attribute with `_` or not) or what?

Reply:
For some fields, the API returns a value by the `level_attr + "_"` key, which is why such a check is necessary here.